packages/cli/src/credentials/credentials-finder.service.ts

```diff
@@ -18,6 +18,37 @@ export class CredentialsFinderService {
 		private readonly roleService: RoleService,
 	) {}
 
+	/**
+	 * Fetches global credentials from the database.
+	 */
+	private async fetchGlobalCredentials(trx?: EntityManager): Promise<CredentialsEntity[]> {
+		const em = trx ?? this.credentialsRepository.manager;
+		return await em.find(CredentialsEntity, {
+			where: { isGlobal: true },
+			relations: { shared: true },
+		});
+	}
+
+	/**
+	 * Merges global credentials with the provided credentials list,
+	 * deduplicating based on credential ID.
+	 */
+	private mergeAndDeduplicateCredentials<T extends { id: string }>(
+		credentials: T[],
+		globalCredentials: CredentialsEntity[],
+		mapGlobalCredential: (cred: CredentialsEntity) => T | null,
+	): void {
+		const credentialMap = new Map(credentials.map((c) => [c.id, c]));
+		for (const globalCred of globalCredentials) {
+			if (!credentialMap.has(globalCred.id)) {
+				const mapped = mapGlobalCredential(globalCred);
+				if (mapped) {
+					credentials.push(mapped);
```

Comment:
Minor: Any particular reason why we're mutating the credentials array here over returning a new one with added credentials? Not that it really matters, I just found that a bit surprising 🤔

Reply:
addressed. thanks for the feedback